### PR TITLE
[deployment] improve breaking changes documentation

### PR DESCRIPTION
**Pull request by labo-flg**
Breaking changes docs shall be improved to clarify our deprecation, our breaking changes and provide clear access to migration guides.

### CODE REVIEW

**Comment by labo-flg** [comment]
cc @helene-nguyen I did not add the deprecation/breaking changes for connectors in 6.2. 
could you try to fit some explanation in this new model ?

A mention in the table, a brief section with a short description and a link to a dedicated guide.

cc @richard-julien @Kedae for awareness

**Comment by helene-nguyen** [comment]
@labo-flg We will have deprecation for 6.4 concerning stream connector splitting, I will add this

**Comment by labo-flg** [comment]
I open for review to document changes in 6.3.

File: docs/deployment/breaking-changes.md [hunk 1]
```diff
@@ -4,53 +4,49 @@ This section lists breaking changes introduced in OpenCTI, per version starting
...
-* Client-python method `client.stix_cyber_observable.promote_to_indicator` is now deprecated
-* New Client-python method `client.stix_cyber_observable.promote_to_indicator_v2` introduced
-
+## OpenCTI 6.2
 
-!!! warning "Discontinued Support"
+### Deprecation
 
-    Please note that the deprecated methods will be permanently removed in OpenCTI 6.5.
+<a id="change-to-the-observable-promote-API"></a>
```
Comment (aHenryJard) [inline docs/deployment/breaking-changes.md:22]: why do you need a a tag ? links are generated

File: docs/deployment/breaking-changes/6.2-promote-to-indicator.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
+# Change to the observable "promote" API
+
+!!! info ""
+
+    * **Deprecated in**: `OpenCTI 6.2`
+    * **Removed in**: `OpenCTI 6.5`
```
Comment (aHenryJard) [inline docs/deployment/breaking-changes/6.2-promote-to-indicator.md:6]: it's a bit strange to be on the future, but I think if we put "sill be removed in" we are going to forget to change in 6.5+

**Review by aHenryJard — COMMENTED** [review]
Looks good except for the links, I think we should keep the generated one

File: docs/deployment/breaking-changes.md (hunk 1 above)
Comment (labo-flg) [reply docs/deployment/breaking-changes.md:22]: actually I've been copy-pasting how it's done elsewhere. Anchor links are not generated it seems.

File: docs/deployment/breaking-changes.md [hunk 3]
```diff
@@ -4,53 +4,49 @@ This section lists breaking changes introduced in OpenCTI, per version starting
...
 
-When `want_assertions_signed` and `want_authn_response_signed` SAML parameter are not present in OpenCTI configuration, 
-the default is now set to `true` by the underlying library (passport-saml) when previously it was `false` by default.
+Upgrading `passport-saml` library introduced a breaking change with respect to the default SAML parameters regarding signing responses and assertions. 
+
+For more details, see [this migration guide](./breaking-changes/6.2-saml-authentication.md).
 
 #### How to migrate
 
 If you have issues after upgrade, you can try with both parameters set to `false`.
```
Comment (labo-flg) [inline docs/deployment/breaking-changes.md:40]:
```suggestion
```

File: docs/deployment/breaking-changes/6.2-saml-authentication.md [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+    * **Changed in**: `OpenCTI 6.2`
+
+## Description of changes
+
+Upgrading `passport-saml` library introduced a breaking change with respect to the default SAML parameters regarding signing responses and assertions.
+
+When `want_assertions_signed` and `want_authn_response_signed` SAML parameter are not present in OpenCTI configuration,
+the default is now set to `true` by `passport-saml` library when previously it was `false` by default.
+
+## Migrate guide
```
Comment (labo-flg) [inline docs/deployment/breaking-changes/6.2-saml-authentication.md:14]:
```suggestion
## Migration guide
```

File: docs/deployment/breaking-changes/6.3-removing-some-connectors.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+The existing connectors affected by this change will be removed and replaced with these two new connectors.
+
+The connectors concerned are:
+- **Tanium stream connector**
+  - Removed: tanium
+  - Replaced by:
+    - tanium-incidents external import connector
+    - tanium-intel stream connector
+- H**arfang Lab stream connector**
```
Comment (labo-flg) [inline docs/deployment/breaking-changes/6.3-removing-some-connectors.md:23]:
```suggestion
- **Harfang Lab stream connector**
```

File: mkdocs.yml [hunk 6]
```diff
@@ -117,7 +117,12 @@ nav:
       - Configuration: deployment/configuration.md
       - Authentication: deployment/authentication.md
       - Upgrade: deployment/upgrade.md
-      - Breaking changes: deployment/breaking-changes.md
+      - Breaking changes:
+        - Index: deployment/breaking-changes.md
+        - Migration guides:
+          - Promote to Indicator: deployment/breaking-changes/6.2-promote-to-indicator.md
+          - SAML authentication: deployment/breaking-changes/6.2-saml-authentication.md
+          - New filter API: deployment/breaking-changes/5.12-filters.md
```
Comment (labo-flg) [inline mkdocs.yml:125]:
```suggestion
          - New filter API: deployment/breaking-changes/5.12-filters.md
          - Removing some connectors: deployment/breaking-changes/6.3-removing-some-connectors.md
```

**Pull request merged**